### PR TITLE
chore: add GitHub Action workflow for main CI checks

### PR DESCRIPTION
**Pull request by mathiasbynens**
This runs all the checks our current Travis setup does, except it only does it on Ubuntu/Linux for now. I’d like to address other OSes in separate patches. PTAL

### CODE REVIEW

**Comment by mathiasbynens** [comment]
Jack, given your experience setting up our current CI, PTAL.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        run: |
+          npm run lint
+          npm run generate-docs
+          npm run ensure-correct-devtools-protocol-revision
+
+      - name: Run unit tests
+        env:
+          CHROMIUM: true
+        run: |
+          xvfb-run --auto-servernum npm run unit
```
Comment (jackfranklin) [inline .github/workflows/main.yml:53]: for my understanding: why do we need xvfb-run now?

**Review by jackfranklin — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (mathiasbynens) [reply .github/workflows/main.yml:53]: GitHub Actions workflows run in a headless environment. We need xvfb on Travis CI as well: https://github.com/puppeteer/puppeteer/blob/9a8479a52a7d8b51690b0732b2a10816cd1b8aef/.travis.yml#L2

Comment (mathiasbynens) [reply .github/workflows/main.yml:53]: Example error log without xvfb: https://github.com/puppeteer/puppeteer/runs/1668553136#step:6:1050

**Pull request merged**